toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/Map.kt

```diff
@@ -30,19 +31,22 @@ import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.viewinterop.AndroidView
 import com.arcgismaps.mapping.ArcGISMap
 import com.arcgismaps.mapping.view.MapView
+import kotlinx.coroutines.launch
 
 /**
  * A compose equivalent of the [MapView].
  *
  * @param modifier Modifier to be applied to the Map
  * @param arcGISMap the [ArcGISMap] to be rendered by this composable
+ * @param onViewPointChanged lambda invoked when the viewpoint of the Map has changed
  * @param overlay the composable overlays to display on top of the Map. Example, a compass, floorfilter etc.
  * @since 200.3.0
  */
 @Composable
 public fun Map(
     modifier: Modifier = Modifier,
     arcGISMap: ArcGISMap? = null,
+    onViewPointChanged: (() -> Unit)? = null,
```

Comment:
Not sure if there is a reason this is initialized to be null, but you could initialize it to a default lambda. 
```suggestion
    onViewPointChanged: () -> Unit = {},
```

Reply:
Defaulting the parameter to null is the correct thing to do. It expresses to the user that the parameter is "absent". See [Compose API guidelines](https://github.com/androidx/androidx/blob/androidx-main/compose/docs/compose-component-api-guidelines.md#nullable-parameter).

Reply:
casing: It''s a "Viewpoint" that is changing, not a "ViewPoint", so the parameter name should be `onViewpointChanged `.

Reply:
We need to consider passing the changed viewpoint to the event callback, so users can immediately process the changed viewpoint. Since there are two types of viewpoints the user might be interested in, we might have to support two different "viewpoint changed events", one passing a viewpoint of type `CenterAndScale` and another passing a viewpoint of type `BoundingGeometry`. We need to think about the naming, but something like this may work:
```kotlin
onViewPointChangedForScale: ((Viewpoint) -> Unit)? = null,
onViewPointChangedForVisibleArea: ((Viewpoint) -> Unit)? = null,
```

Reply:
As per our discussion, I have updated the parameter name